plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/manager/VmwareStorageManagerImpl.java

```diff
@@ -523,7 +523,7 @@ public Answer execute(VmwareHostService hostService, CreateVolumeFromSnapshotCom
 
         String details = null;
         boolean success = false;
-        String newVolumeName = UUID.randomUUID().toString().replaceAll("-", "");
+        String newVolumeName = UUID.randomUUID().toString().replace("-", "");
```

Comment:
we need to use replaceAll here.
The UUID contains 4 "-".

Reply:
@weizhouapache are you sure? according to https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#replace-java.lang.CharSequence-java.lang.CharSequence- the replace method already replaces each accurance. I might be overlooking versions, though.

Reply:
@DaanHoogland a. yes, you are right.